db.go

```diff
@@ -1074,11 +1071,48 @@ func (db *DB) handleFlushTask(ft flushTask) error {
 func (db *DB) flushMemtable(lc *z.Closer) error {
 	defer lc.Done()
 
+	var sz int64
+	var itrs []y.Iterator
+	var mts []*memTable
+	slurp := func() {
```

Comment:
can we check the size before we wait on the channel?

Reply:
I think that is what we are doing, I think I got confused or didn't understand what @harshil-goel meant.